netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,204 @@
+  usageTable.getColumnModel.setColumnSelectionAllowed(false)
+  usageTable.getTableHeader().setReorderingAllowed(false)
+  usageTable.setTableHeader(null)
+  usageTable.setRowMargin(3)
+  usageTable.setRowHeight(20)
+  usageTable.getColumnModel.getColumn(1).setMinWidth(300)
+  usageTable.getColumnModel.getColumn(0).setMinWidth(40)
+  usageTable.setShowGrid(false)
+
+  def init(editorArea: EditorArea) {
+    if(this.editorArea == null) {
+      this.editorArea = editorArea
+      usageTable.addMouseListener(new MouseAdapter() {
+        override def mouseClicked(e: MouseEvent): Unit = {
+          usageBox.setVisible(false)
+          val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+          editorArea.centerCursorInScrollPane(token.start)
+          editorArea.select(token.start, token.end)
+        }
+      })
+
+      usageTable.addKeyListener(new KeyAdapter() {
+        override def keyPressed(e: KeyEvent): Unit = {
+          e.getKeyCode match {
+            case KeyEvent.VK_ESCAPE => usageBox.setVisible(false)
+            case _ => e.setSource(editorArea)
+              editorArea.dispatchEvent(e)
+          }
+        }
+      })
+    }
+  }
+
+  usageTable.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR))
+
+  def showBox(popupLocation: Point, cursorPosition: Int): Unit = {
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), cursorPosition)
+    for {token <- tokenOption} {
+      if(token.tpe == TokenType.Ident || token.tpe == TokenType.Command || token.tpe == TokenType.Reporter) {
+        val tokens = getUsage(editorArea.getText(), token)
+        dataModel.setRowCount(0)
+        for (t <- tokens) {
+          dataModel.addRow(Array[AnyRef](t, editorArea.getLineText(t.start).trim))
+        }
+        if (dataModel.getRowCount != 0) {
+          usageTable.setDefaultRenderer(classOf[String], new LineRenderer(Some(token.text)))
+          usageTable.setPreferredScrollableViewportSize(usageTable.getPreferredSize())
+          usageTable.setFillsViewportHeight(true)
+          usageBox.setSize(usageTable.getPreferredSize)
+          usageTable.validate()
+          usageBox.setLocation(popupLocation)
+          usageBox.setVisible(true)
+        }
+      }
+    }
+  }
+
+  def getUsage(source: String, token: Token): Seq[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    var iter = scala.collection.mutable.Seq[Token]()
+    var prevLineNo = -1
+    val tokenLineNo = editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument], token.start)
+    for(t <- iterator){
+      if( t.text.equalsIgnoreCase(token.text) &&
+        (editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument], t.start)) != tokenLineNo &&
+        editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument], t.start) != prevLineNo) {
```

Comment:
This `for`-comprehension is fine, but for your knowledge it's possible to add the `if` as part of the comprehension, so:
```scala
for (t <- iterator) {
  if (meetsCondition(t)) {
    // some action...
  }
}
```

Becomes:
```scala
for (t <- iterator if meetsCondition(t)) {
  // some action...
}
```
You can do *a lot* in scala with `for`, and it's also OK to use alternatives instead of `for`.